### DISCUSSION TITLE
Use `->can` in routes with Team enabled

### DISCUSSION THREAD

**Discussion by Vito0912** [spatie/laravel-permission#2386] [Q&A]
I am using the `spatie/laravel-permissions` package and am having trouble using the team feature. 

I want to check if a user has certain permissions within a specific team (at route level [api.php]). However, I keep getting an unauthorized message because the team ID is not set in the route. 
But: The team ID is set through middleware (which is called before the route and I can see the print statement being executed with proper id), but when I query the team ID in the route, it is null. 
I would like to avoid validate the team ID/permission in the controller. Can anyone help me solve this issue?

The User has the role/permissions

My `api.php` (excerpt):
```
    Route::group(['prefix' => '', 'middleware' => 'auth'],function () use ($router) {
        print_r('permission_id: ' .  app(\Spatie\Permission\PermissionRegistrar::class)->getPermissionsTeamId());

        $router->get('/info/{region_id}', [AuthController::class, 'info'])->
        where('region_id', UUID_EXPRESSION)->can(PermHelper::CAN_VIEW_USER_INFO);
    });
 ```
 
 My `RegionPermission.php` (called with middleware auth):
 ```
     public function handle(Request $request, Closure $next): Response
    {
        if (!empty(auth()->user())) {
            $region_uuid = $request->route('region_id');
            if($region_uuid == null) {
                $region_uuid = $request->input('region_id');
            }
            if($region_uuid == null)
                $region_uuid = $request->input('regionId');
        }

        if(isset($region_uuid) && $region_uuid != null) {
            $region_id = Region::whereUuid($region_uuid)->value('region_id');
            app(\Spatie\Permission\PermissionRegistrar::class)->setPermissionsTeamId($region_id);
        }


        return $next($request);
    }
 ```
 Note that my $region_id is the corresponding Team ID
 
 Thank you!

**Comment by parallels999** [comment]
```php
if (! empty(auth()->user()) || ! empty(auth('api')->user())) {
```
Did you try that?
[content-working-with-teams-permissions](https://spatie.be/docs/laravel-permission/v5/basic-usage/teams-permissions#content-working-with-teams-permissions)
